Library/Homebrew/formulary.rb

```diff
@@ -906,9 +906,10 @@ def self.tap_formula_name_type(tapped_name, warn:)
     user, repo, name = tapped_name.split("/", 3).map(&:downcase)
     tap = Tap.fetch user, repo
     type = nil
+    alias_name = tap.core_tap? ? name : "#{tap}/#{name}"
 
-    if (possible_alias = tap.alias_table[name].presence)
-      name = possible_alias
+    if (possible_alias = tap.alias_table[alias_name].presence)
```

Comment:
That makes sense. There was some discussion of this inconsistency in another PR that is in review.

Reply:
https://github.com/Homebrew/brew/pull/16623#discussion_r1485424380 if you're curious.

Reply:
I'm was curious, and I totally agree that it makes most sense to make the alias table only contain short names.